src/material-experimental/mdc-sidenav/harness/sidenav-harness.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk-experimental/testing';
+import {SidenavHarnessFilters} from './sidenav-harness-filters';
+
+/**
+ * Harness for interacting with a standard mat-sidenav in tests.
+ * @dynamic
+ */
+export class MatSidenavHarness extends ComponentHarness {
+  static hostSelector = '.mat-sidenav';
```

Comment:
I think this could extend a `MatDrawerHarness` and just add the `isFixedInViewport` check.

Reply:
The reason why I didn't do anything with `mat-drawer` is because we haven't really exposed it anywhere. E.g. we don't have any live examples of it and it's nowhere in the docs from what I can tell.

Reply:
It doesn't have a separate page on our docs site, but it is actually discussed in the [sidenav docs](https://material.angular.io/components/sidenav/overview).